### ISSUE TITLE
No JSON object could be decoded in example Django app

### ISSUE THREAD

**Issue by josefernandez91** [gunthercox/ChatterBot#398]
```python
sudo ./manage.py runserver 127.0.0.1:8000
Performing system checks...

System check identified no issues (0 silenced).
November 08, 2016 - 16:23:10
Django version 1.10.3, using settings 'example_app.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
[08/Nov/2016 16:23:17] "GET / HTTP/1.1" 200 2577
[08/Nov/2016 16:23:17] "GET /static/js/js.cookie.js HTTP/1.1" 200 1615
[08/Nov/2016 16:23:17] "GET /static/js/bootstrap.js HTTP/1.1" 200 36816
[08/Nov/2016 16:23:17] "GET /static/js/jquery.js HTTP/1.1" 200 261746
[08/Nov/2016 16:23:17] "GET /static/css/bootstrap.css HTTP/1.1" 200 141622
Internal Server Error: /api/chatterbot/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/chatterbot/ext/django_chatterbot/views.py", line 28, in post
    data = json.loads(request.read().decode('utf-8'))
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
[08/Nov/2016 16:23:20] "POST /api/chatterbot/ HTTP/1.1" 500 15080
```

**Comment by gunthercox** [comment]
Hi @josefernandez91, I'm looking in to this. My goal is to create a test case that recreates the issue.

**Comment by gunthercox** [comment]
@josefernandez91 Could you tell me what version of ChatterBot you have installed? You should be able to check this by running `python -m chatterbot --version`.

**Comment by maclogan** [comment]
I am having a similar issue where the example will work fine until you refresh the page or let it idle long enough, then it will throw this error. 
`Internal Server Error: /api/chatterbot/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/ChatterBot-0.4.12-py2.7.egg/chatterbot/ext/django_chatterbot/views.py", line 41, in post
    response_data = self.chatterbot.get_response(input_data)
  File "/usr/local/lib/python2.7/dist-packages/ChatterBot-0.4.12-py2.7.egg/chatterbot/chatterbot.py", line 204, in get_response
    statement, response, confidence = self.generate_response(input_statement)
  File "/usr/local/lib/python2.7/dist-packages/ChatterBot-0.4.12-py2.7.egg/chatterbot/chatterbot.py", line 223, in generate_response
    confidence, response = self.logic.process(input_statement)
  File "/usr/local/lib/python2.7/dist-packages/ChatterBot-0.4.12-py2.7.egg/chatterbot/adapters/logic/multi_adapter.py", line 31, in process
    confidence, output = adapter.process(statement)
  File "/usr/local/lib/python2.7/dist-packages/ChatterBot-0.4.12-py2.7.egg/chatterbot/adapters/logic/base_match.py", line 59, in process
    confidence, closest_match = self.get(input_statement)
  File "/usr/local/lib/python2.7/dist-packages/ChatterBot-0.4.12-py2.7.egg/chatterbot/adapters/logic/base_match.py", line 23, in get
    statement_list = self.context.storage.get_response_statements()
  File "/usr/local/lib/python2.7/dist-packages/ChatterBot-0.4.12-py2.7.egg/chatterbot/adapters/storage/storage_adapter.py", line 84, in get_response_statements
    statement_list = self.filter()
  File "/usr/local/lib/python2.7/dist-packages/ChatterBot-0.4.12-py2.7.egg/chatterbot/adapters/storage/django_storage.py", line 75, in filter
    results.append(self.model_to_object(statement_object))
  File "/usr/local/lib/python2.7/dist-packages/ChatterBot-0.4.12-py2.7.egg/chatterbot/adapters/storage/django_storage.py", line 23, in model_to_object
    extra_data=json.loads(statement_model.extra_data, encoding='utf8')
  File "/usr/lib/python2.7/json/__init__.py", line 351, in loads
    return cls(encoding=encoding, **kw).decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded`

If I flush the database and retrain it, it works again.

I have ChatterBot-0.4.12 installed running the newest example and Django 1.10.3.  Hopefully this helps you.

**Comment by maclogan** [comment]
Is there news on a potential solution to this?

**Comment by gunthercox** [comment]
Not yet. For some reason, I can't seem to recreate the error. I will try to look into if further as soon as I have time.

In the mean time, if anyone else feels like taking a crack at this, the issue is coming from these lines [34-37] in https://github.com/gunthercox/ChatterBot/blob/master/chatterbot/ext/django_chatterbot/views.py#L34

```python
        if request.is_ajax():
            input_data = json.loads(request.read().decode('utf-8'))
```

**Comment by gunthercox** [comment]
I believe this issue has been corrected by #428. I'm going to close this ticket off for now, feel free to reopen it if the issue persists.
